### ISSUE TITLE
No validation for transcript's metadata

### ISSUE THREAD

**Issue by kouloumos** [bitcointranscripts/transcription-review-backend#221]
I observed an issue on the UI that has to do with how transcripts' speakers are parsed and displayed.
Looking into it I realize that it arises because of how we validate (or not validate) the metadata that are pushed to the queue.

In this screenshot you can see two transcripts, one shows speakers, the other does not:
![image](https://github.com/bitcointranscripts/transcription-review-front-end/assets/18506343/156afbc7-7240-4a89-a57f-d8f92b709b46)

But in the database, both have speakers, but the speakers data are not stored in the same way and that's from where the issue arises:
```
// The Bitcoin Development Kit
"speakers": [
    "Alekos Filini",
    "Daniela Brozzoni"
  ],
```
```
// FROST Implementers Round Table I
"speakers": "['Christopher Allen', 'Jesse Posner', 'Conrado Gouvêa', 'Ian Goldberg', 'Chelsea Komlo', 'Tim Ruffing']",
```

It seems that the backend is not validating the format of speakers, therefore accepting both types of values. And based on the current content of both staging&production it seems that people have been pushing both formats. 

A weird related behavior, is that the "missing" speakers are displayed in the "Editor" and in the "My Account" page, if the user claims that transcript, which probably means that the speakers are parsed differently in those pages:
![ezgif com-video-to-gif(1)](https://github.com/bitcointranscripts/transcription-review-front-end/assets/18506343/7377d8d7-50af-469d-905f-b05628b75d6b)

For now, we can just fix the formats directly in the database and speakers will be displayed correctly.
But to solve this, I think that we should:
- validate the individual metadata fields in the backend, to avoid this kind of issues in the future

**Comment by kouloumos** [comment]
I think that I might have not understood correctly the source of the issue that I am observing on the UI.

These are the current transcript on production:
![image](https://github.com/bitcointranscripts/transcription-review-backend/assets/18506343/949bf6a4-845a-4d9d-8535-7af80fa980d2)

Based on what I said above, the transcripts for which you can see the speakers, should have the "correct format" in the database, but this is not true. This is the speakers format for the last one:
```
// Challenges in implementing coinjoin in hardware wallets - Pavol Rusnak
"speakers": "['Pavol Rusnak']"
```

But another transcript that has the same format, doesn't display speakers correctly:
```
// Weighing Transactions, The Witness Discount -  Murch - TABConf 2022
"speakers": "['Mark Erhardt']"
```

It seems that the issues is a frontend-backend combination.

**Comment by Extheoisah** [comment]
Good catch. We actually should do some validation. However, it is also expected that tstbtc gives the same structure of data for the metadata. In the case of speakers as you highlighted, tstbtc sometimes formats it as a string instead of an array. The frontend expects an array of speakers and that is why it shows as "N/A" if the backend returns a string. 
In this light, there are three actionable items:
1. fix the metadata results in tstbtc so that we always get a consistent metadata structure.
2. add validation on the backend for metadata to match the expected structure from tstbtc
3. update the current data in the db so the frontend is able to parse the metadata

**Comment by kouloumos** [comment]
>     1. fix the metadata results in tstbtc so that we always get a consistent metadata structure.

Already working on this. tags, categories and speakers will be send as an array. What if there is None? Up until now we were sending `speakers: "None"`. Should we prefer an empty array `speakers: []`?

**Comment by Emmanuel-Develops** [comment]
> Already working on this. tags, categories and speakers will be send as an array. What if there is None? Up until now we were sending speakers: "None". Should we prefer an empty array speakers: []?

Empty array is a good option. 

Regarding why the authors show in the editor but not in the table, I wrote some logic that parses that stringified array to a normal array we can use on the FE, it was mostly a temp solution so it's good we are addressing it now.

**Comment by Extheoisah** [comment]
>1.fix the metadata results in tstbtc so that we always get a consistent metadata structure.
2.add validation on the backend for metadata to match the expected structure from tstbtc
3.update the current data in the db so the frontend is able to parse the metadata


tasks 2 and 3 is complete. moving task 1 to tstbtc repo and closing this
